add codec specs for DVB subtitles (codec ID S_DVBSUB)

mbunkus opened the pull request:
This is the graphical subtitle format used by the Digital Video Broadcasting standard. There are several programs that have been writing Matroska files with such tracks for years. mkvmerge has only gained support for it after release 9.7.1.

This pull request makes that storage method official.

dericed commented on subtitles.md:
```diff
@@ -523,3 +523,32 @@ When TextST subtitles are stored inside Matroska, the only allowed character set
+*    2 bytes: ancillary page ID (bit string, left bit first)
+*    1 byte: subtitling type (bit string, left bit first)
+
+The semantics of these bytes bytes are the same as the ones described in section 6.2.41 "Subtitling descriptor" of ETSI EN 300 468.
```
`bytes bytes`

mbunkus replied:
Meh :) Fixed.

dericed commented on subtitles.md:
```diff
@@ -523,3 +523,32 @@ When TextST subtitles are stored inside Matroska, the only allowed character set
+
+Each Matroska Block consists of one or more DVB Subtitle Segments as described in segment 7.2 "Syntax and semantics of the subtitling segment" of ETSI EN 300 743.
+
+DVB Subtitle Segments are normally shown until a subsequent Subtitle Segment is encountered. Therefore the Matroska Block MAY have no Duration. In that case a player MUST display Subtitle Segments within a Matroska Block until the next Subtitle Segments are encountered.
```
Re: `DVB Subtitle Segments are normally shown until a subsequent Subtitle Segment is encountered.` But if the Block has a duration then it should not be shown until the subsequent Subtitle Segment, right?

dericed replied:
Also I think `SHOULD be displayed` may be better than `are normally shown`.

dericed replied:
Re: `In that case a player MUST display Subtitle Segments`. This makes it seem as if the player does not have the option to not display the subtitle, such as when the subtitle is disabled.

mbunkus replied:
Well, in the codec specs we always describe what a player must do in case it actually displays the track. I don't think we should litter everything with "but only if the track is enabled and selected for display" or something like that.

mbunkus replied:
I can merge this and the following paragraph. How about…

> A Matroska Block MAY have a duration. If it does, a player MUST show the Segments for that Duration or until the next Block for the track is due to be shown, whichever occurs first. If the Block does not have a Duration, a player MUST show the Segments until the next block for the track is due to be shown.

dericed commented on subtitles.md:
```diff
@@ -523,3 +523,32 @@ When TextST subtitles are stored inside Matroska, the only allowed character set
+
+DVB Subtitle Segments are normally shown until a subsequent Subtitle Segment is encountered. Therefore the Matroska Block MAY have no Duration. In that case a player MUST display Subtitle Segments within a Matroska Block until the next Subtitle Segments are encountered.
+
+A multiplexer MAY use a Duration, e.g. by calculating the distance between two subsequent Blocks containing Subtitle Segments. If a Matroska Block has a Duration, a player MUST display that Segment only for the duration of the Block's Duration.
```
I think the logic of these last two paragraphs may be more clear if reordered like below.

> A multiplexer MAY use a Duration, e.g. by calculating the distance between two subsequent Blocks containing Subtitle Segments. If a Matroska Block has a Duration, a player MUST display that Segment only for the duration of the Block's Duration. If a Matroska Block has no Duration, a player MUST display Subtitle Segments within a Matroska Block until the next Subtitle Segments are encountered.

Also, I think `Segment` in this context should be avoided to prevent confusion with the `Segment Element`, using `Subtitle Segment` may be better. Also noting that somethings `Segment` or `Subtitle Segment` is plural and something singular, should be consistent.

mbunkus replied:
I'd rather not say anything about how a multiplexer calculates a duration. Let's only state how a demultiplexer/player must work. That's how most codec specs are written, too.

mbunkus commented:
I've just force-pushed modifications, especially for the handling of the display duration. Please re-review.

robUx4 commented on subtitles.md:
```diff
@@ -523,3 +523,30 @@ When TextST subtitles are stored inside Matroska, the only allowed character set
+
+Each Matroska Block consists of one or more DVB Subtitle Segments as described in segment 7.2 "Syntax and semantics of the subtitling segment" of ETSI EN 300 743.
+
+A Matroska Block MAY have a Duration. If a Matroska Block has a Duration, a player MUST display that Segment only for the duration of the Block's Duration. If a Matroska Block has no Duration, a player MUST display Subtitle Segments within a Matroska Block until the next Subtitle Segments are encountered.
```
Not sure the duration explanation belongs. It's probably how the subtitle (and video) duration should always be handled, as opposed to audio duration for example.

robUx4 replied:
On the other it should say that the DVB duration should be written in the Matroska `Block`.

mbunkus replied:
Yeah, you're probably right. How long a packet should be displayed belongs to the general Matroska specs, not codec-spcific description.

mbunkus commented:
I've removed the section about how to determine how long to display the segments and replaced it with a note stating that each Matroska Block SHOULD have a Duration. Result's been force-pushed.
